k9mail/src/main/java/com/fsck/k9/mailstore/migrations/MigrationTo58.java

```diff
@@ -0,0 +1,31 @@
+package com.fsck.k9.mailstore.migrations;
+
+
+import android.database.sqlite.SQLiteDatabase;
+
+
+class MigrationTo58 {
+    static void cleanUpOrphanedData(SQLiteDatabase db) {
+        cleanUpFtsTable(db);
+        cleanUpMessagePartsTable(db);
+    }
+    
+    private static void cleanUpFtsTable(SQLiteDatabase db) {
+        MigrationTo56.cleanUpFtsTable(db);
+    }
+
+    private static void cleanUpMessagePartsTable(SQLiteDatabase db) {
+        db.execSQL("DELETE FROM message_parts WHERE root NOT IN " +
+                "(SELECT message_part_id FROM messages WHERE deleted = 0)");
```

Comment:
@Valodim 
Sorry, my statement was not correct. The actual problem is, that `row NOT IN (..., NULL, ...)` always gives an empty result. You can test this in your test database using `SELECT * FROM a WHERE a NOT IN (NULL, 2);`. This gives no result, although one could argue, that it should return `5`, because `5!=2` and `5!=NULL`.

That's why the query in `MigrationTo58` is not sufficient (under the assumption that `message_part_id` in `messages` is allowed to be `NULL` -- but this has to be clarified, obviously).